core/include/scipp/core/element/arithmetic.h

```diff
@@ -71,26 +71,42 @@ constexpr auto mod_equals = overloaded{
 
 struct add_types_t {
   constexpr void operator()() const noexcept;
-  using types = arithmetic_and_matrix_type_pairs;
+  using types =
+      decltype(std::tuple_cat(std::declval<arithmetic_and_matrix_type_pairs>(),
+                              std::tuple<std::tuple<time_point, int64_t>,
+                                         std::tuple<int64_t, time_point>>{}));
+};
+
+struct minus_types_t {
+  constexpr void operator()() const noexcept;
+  using types = decltype(std::tuple_cat(
+      std::declval<arithmetic_and_matrix_type_pairs>(),
+      std::tuple<std::tuple<time_point, int64_t>,
+                 std::tuple<time_point, time_point>>{}));
 };
 
 struct times_types_t {
   constexpr void operator()() const noexcept;
   using types = decltype(std::tuple_cat(
       std::declval<arithmetic_type_pairs_with_bool>(),
-      std::tuple<std::tuple<Eigen::Matrix3d, Eigen::Matrix3d>>(),
-      std::tuple<std::tuple<Eigen::Matrix3d, Eigen::Vector3d>>()));
+      std::tuple<std::tuple<Eigen::Matrix3d, Eigen::Matrix3d>>{},
+      std::tuple<std::tuple<Eigen::Matrix3d, Eigen::Vector3d>>{},
+      std::tuple<std::tuple<time_point, int64_t>,
+          std::tuple<int64_t, time_point>>{}));
 };
 
 struct divide_types_t {
   constexpr void operator()() const noexcept;
-  using types = arithmetic_type_pairs;
+  using types = decltype(std::tuple_cat(
+      std::declval<arithmetic_type_pairs>(),
+      std::tuple<std::tuple<time_point, int64_t>,
+          std::tuple<time_point, time_point>>{}));
```

Comment:
See above?
- What is `9:15 am / 17`?
- What is `8 pm / 3 am`?

Reply:
See above